### PR TITLE
Added test of ToU8ForWriting.

### PR DESCRIPTION
**Pull request by zond**

### CODE REVIEW

**Comment by veluca93** [comment]
Giving this to @sboukortt since he had $opinions

File: jxl/src/image.rs [hunk 1]
```diff
@@ -189,7 +189,7 @@ pub mod debug_tools {
 
     impl ToU8ForWriting for u16 {
         fn to_u8_for_writing(self) -> u8 {
-            (self as u32 * 255 / 65535) as u8
+            (self as u32 >> 8) as u8
```
Comment (sboukortt) [inline jxl/src/image.rs:192]: My impression is that what would likely make the most sense is something along the lines of: `((self as u32 * 255 + 32767) / 65535) as u8` (roughly equivalent to converting to a real between 0 and 255 and rounding it). How about it?

File: jxl/src/image.rs [hunk 2]
```diff
@@ -201,7 +201,7 @@ pub mod debug_tools {
 
     impl ToU8ForWriting for u32 {
         fn to_u8_for_writing(self) -> u8 {
-            (self as f32 / (2.0f32.powi(32) - 1.0)).to_u8_for_writing()
+            (self >> 24) as u8
```
Comment (sboukortt) [inline jxl/src/image.rs:204]: Likewise, maybe something like `((self as u64 * 255 + 2147483647) / 4294967295) as u8`.

**Review by sboukortt — APPROVED** [review]

File: jxl/src/image.rs [hunk 3]
```diff
@@ -255,4 +255,56 @@ mod test {
         assert!(image.as_rect().to_pgm().starts_with(b"P5\n32 32\n255\n"));
         Ok(())
     }
+
+    #[cfg(feature = "debug_tools")]
```
Comment (veluca93) [inline jxl/src/image.rs:259]: Might make sense to put these in the debug_tools module (in a further nested `test` submodule)

Comment (zond) [reply jxl/src/image.rs]: Done

**Review by veluca93 — APPROVED** [review]

**Pull request merged**